lightning/src/routing/router.rs

```diff
@@ -1251,9 +1251,10 @@ where L::Target: Logger {
 			|info| info.features.supports_basic_mpp()))
 	} else { false };
 
-	log_trace!(logger, "Searching for a route from payer {} to payee {} {} MPP and {} first hops {}overriding the network graph", our_node_pubkey,
-		maybe_dummy_payee_pk, if allow_mpp { "with" } else { "without" },
-		first_hops.map(|hops| hops.len()).unwrap_or(0), if first_hops.is_some() { "" } else { "not " });
+	log_trace!(logger, "Searching for a route from payer {} to {}payee id {} {} MPP and {} first hops {}overriding the network graph", our_node_pubkey,
+		if payee_node_id_opt.is_none() { "dummy" } else { "" }, maybe_dummy_payee_pk,
```

Comment:
I don't think "dummy" should be in the logging; it's not meaningful to the log consumer. Can we say something like "blinded payee id"?

Reply:
I think that makes it sound like we're logging a blinded payee id, not a dummy value. How about logging `None` for the payee id in the blinded case, and `Some(pk)` in the unblinded case?

Reply:
Hmm... I'm not suggesting logging the dummy value. Rather indicate somehow in the logging that the payee is blinded instead of using the word "dummy".

Reply:
Ah, okay. I don't know if we can conditionally log a pubkey without allocating a `String`, unfortunately 🤔

Reply:
You could make a wrapper on `Option<PublicKey>` that implements `Display` in a conditional manner, IIUC. Since this is trace logging, we can put all the dummy logic in `Display`, not worrying if it's duplicated when trace logging is enabled. Or implement `Display` on `Payee` in lieu of a wrapper, if that makes sense.